lib/Service/SessionService.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2023 Vitor Mattos <vitor@php.rio>
+ *
+ * @author Vitor Mattos <vitor@php.rio>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace OCA\Libresign\Service;
+
+use OCA\Libresign\AppInfo\Application;
+use OCP\IConfig;
+use OCP\ISession;
+
+class SessionService {
+	public function __construct(
+		protected ISession $session,
+		protected IConfig $config,
+	) {
+	}
+
+	public function getSignStartTime(): int {
+		return $this->session->get('libresign-sign-start-time') ?? 0;
+	}
+
+	public function resetDurationOfSignPage(): void {
+		$renewalInterval = (int) $this->config->getAppValue(Application::APP_ID, 'renewal_interval', '0');
+		if ($renewalInterval <= 0) {
+			return;
+		}
```

Comment:
```suggestion
		if ($renewalInterval <= 0) return;
```

Reply:
To use xdebug this isn't too cool because don't will have a explicit return in a step.